packager/signer/signer.go

```diff
@@ -128,13 +128,25 @@ func noRedirects(req *http.Request, via []*http.Request) error {
 	return http.ErrUseLastResponse
 }
 
-func New(cert *x509.Certificate, key crypto.PrivateKey, urlSets []util.URLSet,
+func New(cert *x509.Certificate, key crypto.PrivateKey, docroot string, urlSets []util.URLSet,
 	rtvCache *rtv.RTVCache, shouldPackage func() bool, overrideBaseURL *url.URL,
 	requireHeaders bool) (*Signer, error) {
+	var t http.RoundTripper
+	if docroot != "" {
+		t := &http.Transport{}
+		r := http.NewFileTransport(http.Dir(docroot))
+		t.RegisterProtocol("http", r)
```

Comment:
I think this is correct for the general case (where fetch and sign may differ). Just to clarify, though: [in your case](https://github.com/ampproject/amp-by-example/pull/1592/files#diff-95a5bd8f57d0cf05150691875cb9617b) (where fetch and sign are equal), it's not necessary, as the signed URL [must be https](https://wicg.github.io/webpackage/draft-yasskin-httpbis-origin-signed-exchanges-impl.html#application-signed-exchange).